src/sinks/databend/api.rs

```diff
@@ -0,0 +1,213 @@
+use std::collections::BTreeMap;
+
+use bytes::Bytes;
+use http::Request;
+use http::StatusCode;
+use hyper::Body;
+use serde::{Deserialize, Serialize};
+
+use crate::{http::Auth, http::HttpClient, sinks::util::UriSerde};
+
+use super::error::DatabendError;
+
+#[derive(Serialize, Debug)]
+pub struct StageAttachment {
+    location: String,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    file_format_options: Option<BTreeMap<String, String>>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    copy_options: Option<BTreeMap<String, String>>,
+}
+
+#[derive(Serialize, Debug)]
+pub struct DatabendHttpRequest {
+    sql: String,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    stage_attachment: Option<StageAttachment>,
+}
+
+impl DatabendHttpRequest {
+    pub const fn new(sql: String) -> Self {
+        Self {
+            sql,
+            stage_attachment: None,
+        }
+    }
+
+    pub fn add_stage_attachment(
+        &mut self,
+        location: String,
+        file_format_options: Option<BTreeMap<String, String>>,
+        copy_options: Option<BTreeMap<String, String>>,
+    ) {
+        self.stage_attachment = Some(StageAttachment {
+            location,
+            file_format_options,
+            copy_options,
+        });
+    }
+}
+
+#[derive(Deserialize, Debug)]
+pub struct DatabendHttpResponseSchemaField {
+    pub name: String,
+    pub r#type: String,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct DatabendHttpResponseError {
+    pub code: i64,
+    pub message: String,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct DatabendHttpResponse {
+    pub id: String,
+    pub session_id: Option<String>,
+    pub session: Option<BTreeMap<String, String>>,
+    pub schema: Vec<DatabendHttpResponseSchemaField>,
+    pub data: Vec<Vec<String>>,
+    pub state: String,
+    pub error: Option<DatabendHttpResponseError>,
+    // pub stats: BTreeMap<String, String>,
+    // pub affect: Option<String>,
```

Comment:
If we're not going to use these commented out fields can we remove them until they're needed?

It doesn't look like a few of these are used anywhere, should we wait until there's a need to include them?

Reply:
fixed